api/src/test/java/org/openmrs/module/fhir2/api/dao/impl/FhirLocationDaoImplTest.java

```diff
@@ -101,91 +113,140 @@ public void getLocationByUuid_shouldReturnNullWithUnknownUuid() {
-		Collection<Location> locations = fhirLocationDao.findLocationsByPostalCode(POSTAL_CODE);
+	public void searchForLocations_shouldReturnCorrectLocationByPostalCode() {
+		StringParam postalCode = new StringParam();
+		postalCode.setValue(POSTAL_CODE);
+		Collection<Location> locations = fhirLocationDao.searchForLocations(null, null, null, postalCode, null, null, null,
+		    null);
 		
 		assertThat(locations, notNullValue());
-		assertThat(locations.size(), greaterThanOrEqualTo(2));
+		assertThat(locations.size(), equalTo(2));
 	}
 	
 	@Test
-	public void findLocationsByPostalCode_shouldReturnEmptyCollectionWhenCalledWithUnknownCode() {
-		Collection<Location> locations = fhirLocationDao.findLocationsByPostalCode(UNKNOWN_POSTAL_CODE);
+	public void searchForLocations_shouldReturnEmptyCollectionWhenCalledWithUnknownCode() {
+		StringParam postalCode = new StringParam();
+		postalCode.setValue(UNKNOWN_POSTAL_CODE);
+		Collection<Location> locations = fhirLocationDao.searchForLocations(null, null, null, postalCode, null, null, null,
+		    null);
 		
 		assertThat(locations, notNullValue());
 		assertThat(locations.size(), equalTo(0));
 	}
 	
 	@Test
-	public void findLocationsByState_shouldReturnCorrectLocation() {
-		Collection<Location> locations = fhirLocationDao.findLocationsByState(LOCATION_STATE);
+	public void searchForLocations_shouldReturnCorrectLocationByState() {
+		StringParam state = new StringParam();
+		state.setValue(LOCATION_STATE);
+		Collection<Location> locations = fhirLocationDao.searchForLocations(null, null, null, null, state, null, null, null);
 		
 		assertThat(locations, notNullValue());
-		assertThat(locations.size(), greaterThanOrEqualTo(1));
+		assertThat(locations.size(), equalTo(2));
 	}
 	
 	@Test
-	public void findLocationsByState_shouldReturnEmptyCollectionWhenCalledWithUnknownState() {
-		Collection<Location> locations = fhirLocationDao.findLocationsByState(UNKNOWN_LOCATION_STATE);
+	public void searchForLocations_shouldReturnEmptyCollectionWhenCalledWithUnknownState() {
+		StringParam state = new StringParam();
+		state.setValue(UNKNOWN_LOCATION_STATE);
+		Collection<Location> locations = fhirLocationDao.searchForLocations(null, null, null, null, state, null, null, null);
 		
 		assertThat(locations, notNullValue());
 		assertThat(locations.size(), equalTo(0));
 	}
 	
 	@Test
-	public void findLocationsByTag_shouldReturnLocationsContainingGivenTag() {
+	public void searchForLocations_shouldReturnLocationsContainingGivenTag() {
 		TokenParam locationTag = new TokenParam(FhirConstants.OPENMRS_FHIR_EXT_LOCATION_TAG, LOGIN_LOCATION_TAG_NAME);
-		Collection<Location> locations = fhirLocationDao.findLocationsByTag(locationTag);
+		Collection<Location> locations = fhirLocationDao.searchForLocations(null, null, null, null, null, locationTag, null,
+		    null);
+		
 		assertThat(locations, notNullValue());
-		assertThat(locations.size(), greaterThanOrEqualTo(2));
+		assertThat(locations.size(), equalTo(2));
```

Comment:
Here something like this might work

```java 
		assertThat(locations.iterator().next().getTags().iterator().next().getLocationTagId(),equalTo(LOCATIONT_TAG));
```